backend/editor/entries.py

```diff
@@ -126,11 +131,26 @@ async def get_and_parse_taxonomy(self, uploadfile: UploadFile | None = None):
         try:
             with tempfile.TemporaryDirectory(prefix="taxonomy-") as tmpdir:
                 filepath = await (
-                    self.get_github_taxonomy_file(tmpdir)
+                    self.get_github_taxonomy_file(tmpdir, self.taxonomy_name)
                     if uploadfile is None
                     else self.get_local_taxonomy_file(tmpdir, uploadfile)
                 )
-                await run_in_threadpool(self.parse_taxonomy, filepath)
+                other_filepaths = None
+                if self.taxonomy_name == "ingredients":
+                    other_filepaths = []
+                    for external_taxonomy in (
```

Comment:
In terms of performance, you have here a waterfall of 7 different network calls. How much time does it take to load all the taxonomies? 
How difficult would it be to at least load the external taxonomies concurrently?

Reply:
Good idea, I will launch the network calls concurrently